source/extensions/filters/common/local_ratelimit/local_ratelimit_impl.cc

```diff
@@ -12,7 +14,8 @@ LocalRateLimiterImpl::LocalRateLimiterImpl(
     const std::chrono::milliseconds fill_interval, const uint32_t max_tokens,
     const uint32_t tokens_per_fill, Event::Dispatcher& dispatcher,
     const Protobuf::RepeatedPtrField<
-        envoy::extensions::common::ratelimit::v3::LocalRateLimitDescriptor>& descriptors)
+        envoy::extensions::common::ratelimit::v3::LocalRateLimitDescriptor>& descriptors,
+    bool init_fill_time)
```

Comment:
looks like you always give `init_fill_time = true`. Do we need the param in the first place?

Reply:
No, in local rate limit network filter, `init_fill_time = false`. A test of local rate limit network filter will throw an exception for different time system if we initialize `fill_time_` in `LocalRateLimiterImpl` for local rate limit network filter.

Reply:
Do we know why this is the case? What is the implication of not initializing the fill time at ctor time?

Reply:
Ok, it is a mistake, and I suppose it is because I was not not familiar about the time system in the test before.

There is a assert failure threw by the test shows there is an inconsistent time system. And today I find the problem is only in the test code, not in the source code.

But I cannot fix the test. There is a conflict between the `MockDispatcher` which uses the `GlobalTimeSystem` and the `MockStreamInfo` in `MockReadFilterCallbacks` which uses the `SimulatedTimeSystem`. Do you have some ideas?

Reply:
There is already a TODO(#4160) in `class GlobalTimeSystem`: `Switch default testSystem to SimulatedTimeSystem`. I guess switching to that system might resolve the issue? I don't have context on that TODO though.

If that is a lots of work to do or doesn't work, a quick workaround off top of my head (not necessary to be the final solution): wrapping this line under if condition below:
```
if (!descriptors.empty()) {
    tokens_.fill_time_ = time_source_.monotonicTime();
  }
```
This implicitly limits the this code path to local descriptor override case which is only available in HTTP filter.
It is a bit hacky workaround :) but, IMO, it is still better than having bool flag in the interface all over the places.

Reply:
Unfortunately, I do not think it might work. In fact, a network filter should load time to `fill_time_` while its descriptors are empty.